### DISCUSSION TITLE
hi，大佬，有没有 lsp 或者 cmp 的配置可以支持代码地图？

### DISCUSSION THREAD

**Discussion by ghost** [nshen/learn-neovim-lua#125] [Q&A]
想要一个下面的效果，请问 lsp 或者 cmp 自身支持这个配置吗？还是必须要加插件解决？？

![图片](https://user-images.githubusercontent.com/39549827/195515245-cee0f767-b538-4a19-a32f-91492d853c1e.png)

**Comment by nshen** [comment]
难道是这个插件吗，总看见有人发，我还没用过，大佬可以试试然后分享一下

https://github.com/simrat39/symbols-outline.nvim
https://github.com/stevearc/aerial.nvim

**Reply by ghost** [reply]
能实现的插件倒是挺多的。还有个vista.vim，我就是想问问不用插件能不能实现。

**Reply by HUAHUAI23** [reply]
可以看看lspsaga的源码 里面用lua写的 调用neovim原生的lsp client功能实现
